# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux24/arch-ppc-hotfix.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

PowerPC* define RPA, thus undef itwhile building this amd8111e driver.

  - Rene Rebe <rene@exactcode.de>

is_prep is apparently no longer available in the kernel sources (TODO:
review where it came from) - don't try to use it (for now).

  - Rene Rebe <rene@exactcode.de>

--- linux-2.4.36/drivers/net/amd8111e.h.vanilla	2008-05-11 23:05:41.000000000 +0200
+++ linux-2.4.36/drivers/net/amd8111e.h	2008-05-11 23:09:26.000000000 +0200
@@ -244,6 +244,8 @@
 
 }CMD0_BITS;
 
+#undef RPA	/* PPC */
+
 typedef enum {
 
 	/* VAL3 */
--- linux-2.4.36/drivers/scsi/53c7,8xx.c.vanilla	2008-05-11 23:22:28.000000000 +0200
+++ linux-2.4.36/drivers/scsi/53c7,8xx.c	2008-05-11 23:40:23.000000000 +0200
@@ -1433,7 +1433,8 @@
       command |= PCI_COMMAND_MASTER|PCI_COMMAND_IO;
       pci_write_config_word(pdev, PCI_COMMAND, command);
 
-      if (io_port >= 0x10000000 && is_prep ) {
+#if 0
+      if (io_port >= 0x10000000 && is_prep ) {
 	      /* Mapping on PowerPC can't handle this! */
 	      unsigned long new_io_port;
 	      new_io_port = (io_port & 0x00FFFFFF) | 0x01000000;
@@ -1442,6 +1443,7 @@
 	      pci_write_config_dword(pdev, PCI_BASE_ADDRESS_0, io_port);
 	      pdev->base_address[0] = io_port;
       }
+#endif
     }
 #endif
 
